### PR TITLE
secrets documentation updated

### PR DESCRIPTION
**Pull request by nilminiwso2**
Continuing https://github.com/wso2/docs-ei/pull/1548

### CODE REVIEW

File: en/micro-integrator/docs/develop/creating-artifacts/using_docker_secrets.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Using Docker Secrets
+
+When you deploy your integration solutions in a container environment, you cannot [use the Cipher Tool to encrypt secrets](../../../develop/creating-artifacts/encrypting-synapse-passwords) as you do for a VM deployment.
```
Comment (sachithKay) [inline en/micro-integrator/docs/develop/creating-artifacts/using_docker_secrets.md:3]: It's not that they can't. Its more like an additional capability. 
'WSO2 Micro-integrator comes with a built-in secret repository as a part of its secure vault implementation by default. In addition to this, Micro-integrator also provides built in support for docker secrets and kubernetes secrets for your containerized deployments.
'

File: en/micro-integrator/docs/setup/security/encrypting_plain_text.md [hunk 2]
```diff
@@ -1,118 +1,151 @@
...
+```
+
+Read more about [using dynamic server configurations](../../../setup/dynamic_server_configurations).
+
+To dynamically load secrets to synapse configurations, define the synapse secret as an environment variable.
+For example, consider the JMS user name and password in a proxy service as secrets:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy name="JmsListner" pinnedServers="localhost" startOnLoad="true" transports="http https jms" xmlns="http://ws.apache.org/ns/synapse">
```
Comment (sachithKay) [inline en/micro-integrator/docs/setup/security/encrypting_plain_text.md:145]: This part is wrong. in synapse configs no secrets are always accessed with {wso2:vault-lookup('Alias')}
Example: https://docs.wso2.com/display/ESB480/Working+with+Passwords

File: en/micro-integrator/docs/setup/security/encrypting_plain_text.md [hunk 3]
```diff
@@ -1,118 +1,138 @@
...
-        Alternatively, if you want to retain the password file after the server starts, the file should be named as follows:
+To dynamically load secrets to server configurations, update the parameters in the `deployment.toml` file as an environment variable or system property:
 
-        * For Linux: The file name should be `password-persist`
-        * For Windows: The file name should be `password-persist.txt`
+```toml tab='Environment Variable'
+[keystore.primary]
+password = "$env{ENV_VAR}"
+alias = "$env{ENV_VAR}"
+key_password = "$$env{ENV_VAR}"
```
Comment (sachithKay) [inline en/micro-integrator/docs/setup/security/encrypting_plain_text.md:122]: $$ small error in env variable definition

Comment (nilminiwso2) [reply en/micro-integrator/docs/setup/security/encrypting_plain_text.md]: Fixed

**Pull request merged**